src/terraform/install.sh

```diff
@@ -212,13 +244,23 @@ find_version_from_git_tags TERRAFORM_VERSION 'https://github.com/hashicorp/terra
 find_version_from_git_tags TFLINT_VERSION 'https://github.com/terraform-linters/tflint'
 find_version_from_git_tags TERRAGRUNT_VERSION 'https://github.com/gruntwork-io/terragrunt'
 
+install_terraform() {
+    TERRAFORM_VERSION=$1
+    terraform_filename="terraform_${TERRAFORM_VERSION}_linux_${architecture}.zip"
+    curl -sSL -o ${terraform_filename} "https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/${terraform_filename}"
+}
+
 mkdir -p /tmp/tf-downloads
 cd /tmp/tf-downloads
-
 # Install Terraform, tflint, Terragrunt
 echo "Downloading terraform..."
 terraform_filename="terraform_${TERRAFORM_VERSION}_linux_${architecture}.zip"
-curl -sSL -o ${terraform_filename} "https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/${terraform_filename}"
+install_terraform "$TERRAFORM_VERSION"
+if grep -q "The specified key does not exist." "${terraform_filename}"; then
+    TERRAFORM_VERSION=$(install_prev_vers "terraform" "${TERRAFORM_VERSION}" "https://api.github.com/repos/hashicorp/terraform/releases" | grep "The installed version");
+    TERRAFORM_VERSION=$(echo "$TERRAFORM_VERSION" | sed 's/The installed version: //');
```

Comment:
Let's not use `sed` like this, left one similar comment before.

It is very easy for someone to accidentally update `echo` statements, which would break this.